daemon/cluster/noderunner.go

```diff
@@ -271,7 +271,13 @@ func (n *nodeRunner) enableReconnectWatcher() {
 		if n.stopping {
 			return
 		}
-		config.RemoteAddr = n.cluster.getRemoteAddress()
+		remotes := n.cluster.getRemoteAddressList()
+		if len(remotes) > 0 {
+			config.RemoteAddr = remotes[0]
+		} else {
+			config.RemoteAddr = ""
+		}
```

Comment:
@tonistiigi: The only thing this is used for after the initial startup is finding out the node's local address:

https://github.com/moby/moby/blob/52bded9868f04ec8f06148506666150b017af238/daemon/cluster/cluster.go#L205-L210

I don't think that choosing a random remote for this is any better than using the address that was given to `docker swarm join`. It makes things nondeterministic. The remote does does not need to be up for this to work - it's essentially a routing table lookup.